pkg/engine/fetchit.go

```diff
@@ -368,23 +397,39 @@ func getClone(target *Target) error {
 	} else if !os.IsNotExist(err) {
 		return err
 	}
-
 	if !exists {
 		logger.Infof("git clone %s %s --recursive", target.url, target.branch)
 		if target.pat != "" {
 			target.username = "fetchit"
 			target.password = target.pat
 		}
-		_, err = git.PlainClone(absPath, false, &git.CloneOptions{
+		// default to using existing http method
+		cOptions := &git.CloneOptions{
 			Auth: &githttp.BasicAuth{
 				Username: target.username, // the value of this field should not matter when using a PAT
 				Password: target.password,
 			},
 			URL:           target.url,
 			ReferenceName: plumbing.ReferenceName(fmt.Sprintf("refs/heads/%s", target.branch)),
 			SingleBranch:  true,
-		})
+		}
+		// if using ssh, change auth to use ssh key
+		if target.ssh {
+			logger.Infof("git clone %s ", target.url)
+			authValue, err := ssh.NewPublicKeysFromFile("git", target.sshKey, target.password)
+			if err != nil {
+				logger.Infof("generate publickeys failed: %s", err.Error())
+				return err
+			}
+			cOptions.Auth = authValue
+		}
+		if err := cOptions.Validate(); err != nil {
+			return err
+		}
+		logger.Infof("absPath is: %v", absPath)
+		_, err = git.PlainClone(absPath, false, cOptions)
 		if err != nil {
+			logger.Infof("plainclone is returning error")
```

Comment:
line #429 & #432 are debug lines, right? you can remove those b4 merge
With this plus the `cat` instead of `echo` issue in the workflow, this will be ready to go, thank you!

Reply:
Updated with unnecessary lines removed